### ISSUE TITLE
Design Notification model and edit friending endpoints to make use of it

### ISSUE THREAD

**Issue by omarelkashef** [uchicago-cs/chigame#127]
We need a Notification model that will have updates for the user such as friend requests. A user need to be able to mark notification as read. The model should be general enough to handle all types of notifications. Notification can be marked notification as read, unread or deleted. The model will have another model instance as an "actor" who creates the notification.

**Comment by omarelkashef** [comment]
We also need to update the friending endpoints in #108 to be integrating with the new model.

**Comment by omarelkashef** [comment]
We also designed a custom queryset class for the new model in order to provide some helper methods when querying the model.

**Comment by omarelkashef** [comment]
We decided that each instance of FriendInvitation will have its own notification object even if the invitation is between the same sender and receiver.

**Comment by omarelkashef** [comment]
Closing statment: we created a Notification model that stores updates such as friend request for the users. The user is able to mark a notification as read, unread or deleted.  We also created a custom queryset class to remove redundancy in querying Notification. The friending viewpoints have been modified to make use of the new Notification model.

@omarelkashef  worked on implementing the Notification model and its queryset
@24raniwalar  worked on using the new model in the friending endpoints.
